### PR TITLE
Switch from `jsonpath_lib` to `jsonpath-rust`

### PR DESCRIPTION
**Pull request by illia-bobyr**
## Motivation
`jsonpath_lib` is unmaintained, which is not ideal.  It is also forcing everyone to enable a `preserve_order` feature in the `serde_json` dependency.

`preserve_order` is not following the proper feature model, where features are only adding functionality.  It is actually changing how `serde_json` works, switching from `BTreeMap` to `IndexMap`.

## Solution
Switch from [`jsonpath_lib`](https://crates.io/crates/jsonpath_lib) to [`jsonpath-rust`](https://crates.io/crates/jsonpath-rust).

Change has been tested with both unit and integration tests.

### CODE REVIEW

File: kube-client/src/client/auth/mod.rs [hunk 1]
```diff
@@ -474,22 +475,32 @@ fn token_from_gcp_provider(provider: &AuthProviderConfig) -> Result<ProviderToke
...
+                 Error: {err}"
+            ))
+        })?
+    };
 
-        Err(e) => Err(Error::AuthExec(format!("Could not extract JSON value: {e:}"))),
+    let res = parsed_path.find_slice(json);
 
-        _ => Err(Error::AuthExec(format!("Target value {pure_path:} not found"))),
+    let Some(res) = res.into_iter().next() else {
```
Comment (gregcusack) [inline kube-client/src/client/auth/mod.rs:491]: why use `next()` instead of just calling for the first element directly with res[0]?

File: kube-client/src/client/auth/mod.rs [hunk 2]
```diff
@@ -474,22 +475,32 @@ fn token_from_gcp_provider(provider: &AuthProviderConfig) -> Result<ProviderToke
...
+            "Target {context:?} value {path:?} not found"
+        )));
+    };
+
+    if let Some(val) = res.as_str() {
+        Ok(val.to_owned())
+    } else {
+        Err(Error::AuthExec(format!(
+            "Target {:?} value {:?} is not a string: {:?}",
+            context, path, *res
```
Comment (gregcusack) [inline kube-client/src/client/auth/mod.rs:502]: for cleanliness sake, may want to format this string the same way as the other two above: `{context:?}`, `{path:?}`, `{*res:?}`

**Review by gregcusack — COMMENTED** [review]
ya looks good to me. just a couple comments. thanks for updating this

File: kube-client/src/client/auth/mod.rs (hunk 1 above)
Comment (illia-bobyr) [reply kube-client/src/client/auth/mod.rs:491]: [`find_slice()`](https://docs.rs/jsonpath-rust/0.3.4/jsonpath_rust/struct.JsonPathInst.html#method.find_slice) returns a `Vec<JsonPtr<'a, Value>>`, which can be empty.

So, I would need to check if it is empty first.  And only then it is safe to access `res[0]`.  So, it is a bit less safe, as the check for length is separate from the access. 

I can call `res.get(0)` to get an `Option<&JsonPtr<'a, Value>>`.

But, in both cases, I would get a reference to the contained `JsonPtr`, as the vectors still owns the value.
And it is not what I really want.
I do not need the vector after this point, and I do not want a reference.

I want the first element, by value.
`res.into_iter().next()` is the most accurate way to say that.
It either gives me the first element by value, or `None`.

File: kube-client/src/client/auth/mod.rs (hunk 2 above)
Comment (illia-bobyr) [reply kube-client/src/client/auth/mod.rs:502]: `{*res:?}` is invalid.
Inline names can not contain expressions, only symbols, optionally path qualified.

As soon as a single name is not inlined, it is clearer not to inline any of them.  Mix of inlined and non-lined names can be hard to follow.

File: kube-client/src/client/auth/mod.rs (hunk 1 above)
Comment (gregcusack) [reply kube-client/src/client/auth/mod.rs:491]: ahhh got it. ya makes sense

File: kube-client/src/client/auth/mod.rs (hunk 2 above)
Comment (gregcusack) [reply kube-client/src/client/auth/mod.rs:502]: oh ya ok did not know that. good to know. thank you for the clarifications!
